websites/S/SoloLearn/presence.ts

```diff
@@ -1,174 +1,164 @@
+        }
+      }
 
-    if (name) {
-      if (course) {
-        data.state = `${name} | ${course}`;
-      } else {
-        data.state = name;
+      if (path.includes("/Leaderboard/")) {
+        data.details = "Browsing Leaderboard...";
+        data.state = stripCourse(getElement(".nameTitle"));
       }
-    }
-  }
 
-  if (path.match("/Course")) {
-    data.details = "Viewing Course";
+      if (path.includes("/Blog/")) {
+        data.details = "Browsing Blogs...";
 
-    const name = getElement(".courseDescription > h1");
-    if (name) {
-      data.state = name;
-    }
-  }
+        if (document.querySelector(".post")) {
+          data.details = "Browsing Blog...";
+          data.state = getElement(".articleTitle");
+        }
+      }
 
-  if (path.match("/Courses")) {
-    data.details = "Viewing Courses";
-  }
+      if (path.includes("/Course/")) {
+        data.details = "Browsing Course...";
+        data.state = getElement(".courseDescription > h1");
+      }
 
-  if (path.match("/Codes")) {
-    data.details = "Viewing Codes";
+      if (path.includes("/Profile/")) {
+        data.details = "Browsing Profile...";
 
-    const tab = getElement(".tab.active");
-    if (tab) {
-      data.state = tab;
+        const course = getElement(".course .name");
+        data.state = getElement(".user .name");
+        data.state += course ? ` (${stripCourse(course)})` : "";
+      }
     }
   }
 
-  if (host.match("code.sololearn.com")) {
-    data.details = "Viewing Code";
-
-    const name = getElement(".codeName");
-    if (name) {
-      data.state = name;
+  if (showCourses) {
+    if (path.includes("/Play/")) {
+      const icon: HTMLImageElement = document.querySelector(".content > .icon");
+      data.details = `Learning ${stripCourse(icon.alt)}`;
+      data.state = getElement(".title");
     }
   }
 
-  if (path.match("/Discuss")) {
-    data.details = "Viewing Discussions";
-
-    const name = getElement(".question .header");
-    if (name) {
-      data.details = "Viewing Discussion";
-      data.state = name;
-    }
-
-    const tab = getElement(".tab.active");
-    if (tab) {
-      data.state = tab;
+  if (showCodes) {
+    if (host === "code.sololearn.com") {
+      data.details = "Viewing Code...";
+      data.state = `${getElement(".codeName")} (${getElement(
+        ".tab-box.active"
+      )})`;
     }
   }
 
-  if (path.match("/Leaderboard")) {
-    data.details = "Viewing Leaderboard";
-
-    const type = getElement(".nameTitle");
-    if (type) {
-      data.state = type;
+  if (data.details) {
+    if (data.details.match("(Browsing|Viewing)")) {
+      data.smallImageKey = "reading";
+      data.smallImageText = (await strings).browse;
     }
-  }
-
-  if (path.match("/Blog")) {
-    data.details = "Viewing Blog";
-
-    const name = getElement(".articleTitle");
-    if (name) {
-      data.state = name;
+    if (!showTimestamps) {
+      data.startTimestamp = undefined;
+      data.endTimestamp = undefined;
```

Comment:
`delete data.startTimestamp` instead of setting it to undefined.

Reply:
Deleting is not efficient because it changes the structure of an object, while setting it to undefined releases the content but doesn't force the object to modify its structure.

Reply:
IMO you shouldnt send anything that doesnt need sending.

Reply:
@Alanexei Actually, when you set it to undefined it is never deleted and it's not the same as if the value was never set. Sure it does modify the structure and that's the point, because the property is better off destroyed than just modified.
While it may seem illogical, it is mostly Discord to be blamed since I guess they check for the type in some different way that still keeps the value defined even when you set it to `undefined`, same applies to `null` so delete is the best choice, we could find a workaround on the extension level (as usual) of course but for now that's how it is.

Reply:
Also to back what Bas said. Since it holds value when just set as undefined, you're still sending something which slightly affects performance.

Reply:
My point is to not delete the value, it is to reset the value. It doesn't really matter either way, because the presence behaves as it should. But I will change it to not cause any more fuss. If you are enforcing the `delete` keyword, you should have asked me to put it in the guidelines.

Reply:
We've discussed it before.